### PR TITLE
Add hm1 and hm2

### PR DESCRIPTION
**Pull request by nazaroviktor2**

### CODE REVIEW

**Review by siriusdevs — CHANGES_REQUESTED** [review]
could you please provide readme file?
users definetely need some instructions to start your project, dont you think?
see max228 readme for bad example.


I also see you empty tests :eyes:
django has nice testing submodules right there for you, use them.

**Comment by nazaroviktor2** [comment]
I don't have read.me in the root directory because that's where all my projects are.  The root directory is not a project.  But there is a read.me in each of the project folders to help the user get the project up and running.  If the user needs help with supported requests, then he must launch the application and go to the main page, where he will see the documentation for working with the application.

**Comment by nazaroviktor2** [comment]
> I also see you empty tests eyes django has nice testing submodules right there for you, use them.

Yes django has nice testing submodules. But my django app ***works with image api***. If this server is not running, django will return errors. And git workflows does not start the server with the image api. So I'll create tests for the app, ***I'll get a red X*** on git because the tests will fail

File: django_nasa/django_nasa/urls.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    Function views:
+        1. Add an import:  from my_app import views.
+        2. Add a URL to urlpatterns:  path('', views.home, name='home')
+    Class-based views
+        1. Add an import:  from other_app.views import Home
+        2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
+    Including another URLconf
+        1. Import the include() function: from django.urls import include, path
+        2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
+"""
```
Comment (siriusdevs) [inline django_nasa/django_nasa/urls.py]: excuse-moi, what is this in the name of God and Holy Django the Great?

File: django_nasa/django_nasa/wsgi.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"""
+WSGI config for django_nasa project.
+
+It exposes the WSGI callable as a module-level variable named ``application``.
+
+For more information on this file, see
+https://docs.djangoproject.com/en/4.1/howto/deployment/wsgi/
+"""
```
Comment (siriusdevs) [inline django_nasa/django_nasa/wsgi.py]: again?

File: django_nasa/nasa/admin.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+"""Admin panel function."""
```
Comment (siriusdevs) [inline django_nasa/nasa/admin.py]: is there no admin panel in your project?
do you need an empty admin.py file though? is it compulsory to have it? (idk didnt try)

File: django_nasa/nasa/models.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+"""Models of nasa api."""
```
Comment (siriusdevs) [inline django_nasa/nasa/models.py]: empty again

File: django_nasa/nasa/tests.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+"""Some tests for nasa."""
```
Comment (siriusdevs) [inline django_nasa/nasa/tests.py]: empty again

File: django_nasa/nasa/views.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
+"""Some function for nasa entrypoints."""
+from django.http import JsonResponse
+from django.shortcuts import render, redirect
+import requests as req
+from django.core.handlers.wsgi import WSGIRequest
+from django_nasa.settings import IMAGE_API_TOKEN, IMAGE_API_URL
+
+
+# Create your views here.
```
Comment (siriusdevs) [inline django_nasa/nasa/views.py:11]: well, i do not think i will :smile:

File: django_nasa/templates/mars.html [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+  // Send an HTTP GET request to the Django server with the selected parameters
+  const sendRequest = (params) => {
+  loading.style.display = "block";
+    const url = "/";
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", url + "?" + params, true);
+    xhr.send();
+    xhr.onload = () => {
+
+      if (xhr.status === 200) {
```
Comment (siriusdevs) [inline django_nasa/templates/mars.html:194]: that is some sirius js shit right there

File: django_nasa/templates/mars.html [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+  userBtn.addEventListener("click", () => {
+window.location.href = "/user";
+  });
+
+  okBtn.addEventListener("click", () => {
+    const params = `source=mars&date=${dateInput.value}`;
+    sendRequest(params);
+  });
+
```
Comment (siriusdevs) [inline django_nasa/templates/mars.html:242]: well, i am okay with that, by i suggest you show it to Иван Вячеславович.

Comment (siriusdevs) [reply django_nasa/templates/mars.html:242]: i am more than sure that there are MAP funcitons in js. and a lot more to make this code shrink to a tiny little script

File: image_api/http_config.py [hunk 9]
```diff
@@ -0,0 +1,6 @@
+"""Some http constants for image api."""
```
Comment (siriusdevs) [inline image_api/http_config.py:1]: is there no way you return bad request code?

File: image_api/init.sql [hunk 10]
```diff
@@ -0,0 +1,10 @@
+create  extension if not exists  "uuid-ossp";
```
Comment (siriusdevs) [inline image_api/init.sql:1]: you do not need uuid extension just to mention uuid type. you need it to generate uuid value

File: image_api/tests/test_user_image.py [hunk 11]
```diff
@@ -0,0 +1,5 @@
+II = 1
+
+
+def test_for_test():
+    assert II == 1
```
Comment (siriusdevs) [inline image_api/tests/test_user_image.py]: explain

**Review by siriusdevs — CHANGES_REQUESTED** [review]
+ not enough tests.

you said that it is impossibru to run them, but i am not buying it. you chose this way yourself, stick to it
i believe in you :crossed_fingers: :four_leaf_clover:

**Pull request merged**